### PR TITLE
Style Progress Indicator

### PR DESCRIPTION
**Pull request by diversemix**
#### Background

Themes the Progress Indicator as per the styleguide. Open a second ticket for the white cross on the previous steps. 

#### Any relevant tickets
#224 

#### How has this been tested?
Locally

### CODE REVIEW

**Comment by jsms90** [comment]
After you've resolved merge conflicts, is this ready for review? Or what work still needs to be done for this to come out of WIP?

File: client/elife-theme/src/elements/index.js [hunk 1]
```diff
@@ -1,3 +1,11 @@
 export { default as Button } from './Button'
 export { default as Heading } from './Heading'
 export * from './Heading'
+
+export {
+  default as StepsRoot,
```
Comment (tamlyn) [inline client/elife-theme/src/elements/index.js]: Why not name the export `StepsRoot` then you can just `export *`?

File: client/elife-theme/src/elements/Steps.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  }
+  return th('colorFurniture')
+}
+
+export const Step = css`
+  height: 20px;
+  width: 20px;
+  background-color: ${({ isCurrent, isPast }) => stepCenter(isCurrent, isPast)};
+  border-color: ${({ isCurrent, isPast }) =>
+    isCurrent || isPast ? th('colorPrimary') : th('colorFurniture')};
```
Comment (tamlyn) [inline client/elife-theme/src/elements/Steps.js]: Oh this is neat. I didn't know you could return an interpolated function from and interpolated function.

File: client/elife-theme/src/elements/index.js (hunk 1 above)
Comment (diversemix) [reply client/elife-theme/src/elements/index.js]: Isn't `export *` the spawn of Satan?

Comment (tamlyn) [reply client/elife-theme/src/elements/index.js]: This is a re-export. It's only exporting the things which were imported from `Steps`. JavaScript doesn't have a way to export all variables (for the very good reason you mention).

**Comment by jsms90** [comment]
Looks greyed out for me on the review branch: http://review-214-progre-81p563.gateway.xpub.semioticsquares.com/submit

image

File: client/elife-theme/src/elements/ui/Steps.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { css } from 'styled-components'
+import { th } from '@pubsweet/ui-toolkit'
+
+export default css`
+  margin-bottom: 100px;
```
Comment (g-sam) [inline client/elife-theme/src/elements/ui/Steps.js]: is this the value from the invision prototypes?

File: client/elife-theme/src/elements/ui/Steps.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+`
+
+const stepCenter = (isCurrent, isPast) => {
+  if (isCurrent) {
+    return '#fff'
```
Comment (g-sam) [inline client/elife-theme/src/elements/ui/Steps.js]: Could this be theme.backgroundColor?

File: client/elife-theme/src/elements/ui/Steps.js (hunk 3 above)
Comment (diversemix) [reply client/elife-theme/src/elements/ui/Steps.js]: nope - its a guess

File: client/elife-theme/src/elements/ui/Steps.js (hunk 4 above)
Comment (diversemix) [reply client/elife-theme/src/elements/ui/Steps.js]: good point

File: client/elife-theme/src/elements/ui/Steps.js (hunk 3 above)
Comment (hdrury1) [reply client/elife-theme/src/elements/ui/Steps.js]: an easily changeable guess (more of these plz :) )

Comment (g-sam) [reply client/elife-theme/src/elements/ui/Steps.js]: but i think the correct value is annotated?

Comment (g-sam) [reply client/elife-theme/src/elements/ui/Steps.js]: yes, it's 48px

**Comment by g-sam** [comment]
This needs linking to a PR to pubsweet, right?

**Comment by diversemix** [comment]
This is linked to PubSweet releasing - https://gitlab.coko.foundation/pubsweet/pubsweet/merge_requests/316

**Comment by diversemix** [comment]
Yeah @g-sam / @jsms90 ... this is why its grey ... https://gitlab.coko.foundation/pubsweet/pubsweet/merge_requests/316

**Comment by hdrury1** [comment]
Could I make a suggestion that when we're waiting for PubSweet MRs to go through we make a note on the GitHub PR what this will add/alter/change? i.e. in this case I guess the PubSweet MR will allow us to change the colour to blue, remove the step numbers, add the tick icon?

**Comment by diversemix** [comment]
Good point - this will make the changes apart from the tick icon. Separate issue for tick icon #284

**Pull request merged**